phpBB/includes/acp/acp_search.php

```diff
@@ -76,69 +147,67 @@ function settings($id, $mode)
 			'search_store_results'			=> 'integer',
 		];
 
-		$search = null;
 		$search_options = '';
 
-		foreach ($search_types as $search)
+		foreach ($this->search_backend_collection as $search)
 		{
 			// Only show available search backends
 			if ($search->is_available())
 			{
 				$name = $search->get_name();
-				$type = get_class($search);
+				$type = $search->get_type();
 
-				$selected = ($config['search_type'] == $type) ? ' selected="selected"' : '';
+				$selected = ($this->config['search_type'] == $type) ? ' selected="selected"' : '';
 				$identifier = substr($type, strrpos($type, '\\') + 1);
 				$search_options .= "<option value=\"$type\"$selected data-toggle-setting=\"#search_{$identifier}_settings\">$name</option>";
 
 				$vars = $search->get_acp_options();
 
 				if (!$submit)
 				{
-					$template->assign_block_vars('backend', array(
+					$this->template->assign_block_vars('backend', [
 						'NAME' => $name,
 						'SETTINGS' => $vars['tpl'],
 						'IDENTIFIER' => $identifier,
-					));
+					]);
 				}
 				else if (is_array($vars['config']))
 				{
 					$settings = array_merge($settings, $vars['config']);
 				}
 			}
 		}
-		unset($search);
 
-		$cfg_array = (isset($_REQUEST['config'])) ? $request->variable('config', array('' => ''), true) : array();
-		$updated = $request->variable('updated', false);
+		$cfg_array = (isset($_REQUEST['config'])) ? $this->request->variable('config', ['' => ''], true) : [];
+		$updated = $this->request->variable('updated', false);
 
-		foreach ($settings as $config_name => $var_type)
+		foreach ($settings as $this->config_name => $var_type)
 		{
-			if (!isset($cfg_array[$config_name]))
+			if (!isset($cfg_array[$this->config_name]))
 			{
 				continue;
 			}
 
 			// e.g. integer:4:12 (min 4, max 12)
 			$var_type = explode(':', $var_type);
 
-			$config_value = $cfg_array[$config_name];
-			settype($config_value, $var_type[0]);
+			$this->config_value = $cfg_array[$this->config_name];
```

Comment:
I think you had some accidental replacements in here ;)
```suggestion
			$config_value = $cfg_array[$config_name];
```

Reply:
Yes, I did a mass replacement for the properties, sorry  :sweat_smile: